app/js/components/identifiers.jsx

```diff
@@ -177,34 +155,36 @@ export default class Identifiers extends React.Component {
             `patient/${this.state.patient_uuid}/identifier/${this.state.preffered_identifier_uuid}`
           )
             .then((response) => {
-              (response.error) ?
-                toastr.error(response.error.message)
-                :
-                toastr.success("Identifier Created Successfully.") && this.reload();
+                if(response.error){
+                   this.setState({addErrors: {error: response.error.message}})
+                   }
+                  else{
+                    this.setState({ newState: false });
+                    toastr.success("Identifier Created Successfully.") && this.reload();}
             })
             .catch(error => toastr.error(error))
         })
         .catch(error => toastr.error(error));
     } else {
       apiCall(new_addIdentifiers, 'post', `patient/${this.state.patient_uuid}/identifier`)
         .then((response) => {
-          (response.error) ?
-            toastr.error(response.error.message)
-            :
-            toastr.success("Identifier Created Successfully.") && this.reload();
+            if(response.error){
+               this.setState({addErrors: {error: response.error.message}})
+               }
+              else{
+                this.setState({ newState: false });
+                toastr.success("Identifier Created Successfully.") && this.reload();}
         })
         .catch(error => toastr.error(error));
     }
   }
 
-  /**
-   *
-   * @param {*} e
-   *called on save when editing an identifier
-   */
   callSave(e) {
-    this.setState({ editState: false })
+      this.setState({editErrors: {uuid: '',
```

Comment:
This could be on same line or we could have it on different line but maintaining the amount of white space